maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java

```diff
@@ -526,8 +578,39 @@ private void commands(CliRequest cliRequest) {
 
     // Needed to make this method package visible to make writing a unit test possible
     // Maybe it's better to move some of those methods to separate class (SoC).
-    void properties(CliRequest cliRequest) {
-        populateProperties(cliRequest.commandLine, cliRequest.systemProperties, cliRequest.userProperties);
+    void properties(CliRequest cliRequest) throws ExitException {
+        try {
+            populateProperties(cliRequest, cliRequest.systemProperties, cliRequest.userProperties);
+
+            StringSearchInterpolator interpolator =
```

Comment:
Any reason not to use the `StringRegexInterpolator`?

Reply:
The interpolator thing is pretty much coming from master. What is the difference?

Reply:
I remember that one has superseded the other, I don't exactly remember.

Reply:
As I see in maven codebase use of StringSearch vs Regexp interpolator is 4:5

Reply:
I wonder they there are two what the difference is.

Reply:
Well, I remember the interface https://github.com/codehaus-plexus/plexus-interpolation/blob/master/src/main/java/org/codehaus/plexus/interpolation/Interpolator.java#L24 :smile: